src/components/ClientForm/index.jsx

```diff
@@ -3,14 +3,19 @@ import { Link } from 'react-router-dom';
 import { bool } from 'prop-types';
 import { addYears } from 'date-fns';
 import SpeedDialAction from '@material-ui/lab/SpeedDialAction';
-import { withStyles } from 'material-ui/styles';
-import List, { ListItem, ListItemText } from 'material-ui/List';
-import TextField from 'material-ui/TextField';
-import Tooltip from 'material-ui/Tooltip';
-import Button from 'material-ui/Button';
-import Typography from 'material-ui/Typography';
-import Switch from 'material-ui/Switch';
-import { FormGroup, FormControlLabel } from 'material-ui/Form';
+import { withStyles } from '@material-ui/core/styles';
+import {
```

Comment:
Oh yeah? I thought the goal was to put all of the components at the root level so that users don't find themselves going back and forth with the docs to learn the import path.

Reply:
https://material-ui.com/guides/minimizing-bundle-size/#how-to-reduce-the-bundle-size-

Reply:
TLDR; you'll need to test if tree-shaking is working properly by inspecting the build sizes before and after moving to root imports vs. file-specific imports.

Reply:
That may also explain the memory issue: if we are trying to bundle all of material-ui but run out of memory in the process.

Reply:
Actually this PR was deployed successfully on netlify up until I rebased to include Hooks hehe.

Reply:
### Root imports

```bash
                                                      Asset       Size                         Chunks                    Chunk Names
                            index.0d00b65a601e6dee0ff8.v1.js    1.88 MB                          index  [emitted]  [big]  index
      roboto-latin-300.a1471d1d6431c893582a5f6a250db3f9.woff    19.9 kB                                 [emitted]
      roboto-latin-400.bafb105baeb22d965c70fe52ba6b49d9.woff    19.8 kB                                 [emitted]
     roboto-latin-400.5d4aeb4e5f5ef754e307d7ffaef688bd.woff2    15.3 kB                                 [emitted]
      roboto-latin-500.de8b7431b74642e830af4d4f4b513ec9.woff      20 kB                                 [emitted]
     roboto-latin-500.285467176f7fe6bb6a9c6873b3dad2cc.woff2    15.6 kB                                 [emitted]
                     AuthCallback.d74b0744e789bf1e69c6.v1.js  630 bytes                   AuthCallback  [emitted]         AuthCallback
                          Clients.d66a280386b9c5da2b71.v1.js    1.44 kB                        Clients  [emitted]         Clients
               Clients.ViewClient.e271ca9840f02211bf55.v1.js     300 kB             Clients.ViewClient  [emitted]  [big]  Clients.ViewClient
              Clients.ViewClients.235cf30d65d3fb9d4998.v1.js     221 kB            Clients.ViewClients  [emitted]         Clients.ViewClients
                Clients.ViewScope.24327d1a7dcc7763fbd7.v1.js     196 kB              Clients.ViewScope  [emitted]         Clients.ViewScope
               Clients.ViewScopes.4cd42c769a64bc78cda9.v1.js     194 kB             Clients.ViewScopes  [emitted]         Clients.ViewScopes
                        Dashboard.8932261b505f71932442.v1.js    61.8 kB                      Dashboard  [emitted]         Dashboard
                    Documentation.4264ea8b8ef49ddaa96c.v1.js    60.8 kB                  Documentation  [emitted]         Documentation
                             Home.ddc88b92bc19b8ce922f.v1.js      25 kB                           Home  [emitted]         Home
                            Hooks.3bc2497a3981ecbf9cd4.v1.js    1.43 kB                          Hooks  [emitted]         Hooks
                  Hooks.ListHooks.b741243b02a84b5d1dd8.v1.js     128 kB                Hooks.ListHooks  [emitted]         Hooks.ListHooks
                   Hooks.ViewHook.171e73cb20fb517b9465.v1.js     567 kB                 Hooks.ViewHook  [emitted]  [big]  Hooks.ViewHook
                         NotFound.d7895105b0b0944c98b4.v1.js    62.4 kB                       NotFound  [emitted]         NotFound
                     Provisioners.f82dc17adc223d3b398b.v1.js    1.89 kB                   Provisioners  [emitted]         Provisioners
    Provisioners.ViewProvisioners.c60a314d817e16dcdc8d.v1.js     129 kB  Provisioners.ViewProvisioners  [emitted]         Provisioners.ViewProvisioners
          Provisioners.ViewWorker.e499cb8caaf4691f4d2c.v1.js     243 kB        Provisioners.ViewWorker  [emitted]         Provisioners.ViewWorker
     Provisioners.ViewWorkerTypes.9fac225b7600f80c79d5.v1.js     227 kB   Provisioners.ViewWorkerTypes  [emitted]         Provisioners.ViewWorkerTypes
         Provisioners.ViewWorkers.c05600a150db749bac1a.v1.js     210 kB       Provisioners.ViewWorkers  [emitted]         Provisioners.ViewWorkers
                            Roles.985f69e935c75ed84d5a.v1.js    1.42 kB
```

### File specific imports

```bash
                                                       Asset       Size                         Chunks                    Chunk Names
                            index.ebf2adf444100efb5eab.v1.js    1.49 MB                          index  [emitted]  [big]  index
      roboto-latin-300.a1471d1d6431c893582a5f6a250db3f9.woff    19.9 kB                                 [emitted]
      roboto-latin-400.bafb105baeb22d965c70fe52ba6b49d9.woff    19.8 kB                                 [emitted]
     roboto-latin-400.5d4aeb4e5f5ef754e307d7ffaef688bd.woff2    15.3 kB                                 [emitted]
      roboto-latin-500.de8b7431b74642e830af4d4f4b513ec9.woff      20 kB                                 [emitted]
     roboto-latin-500.285467176f7fe6bb6a9c6873b3dad2cc.woff2    15.6 kB                                 [emitted]
                     AuthCallback.d74b0744e789bf1e69c6.v1.js  630 bytes                   AuthCallback  [emitted]         AuthCallback
                          Clients.d66a280386b9c5da2b71.v1.js    1.44 kB                        Clients  [emitted]         Clients
               Clients.ViewClient.696d4844b8b005e1fc52.v1.js     551 kB             Clients.ViewClient  [emitted]  [big]  Clients.ViewClient
              Clients.ViewClients.175379613d5acb8a2d5c.v1.js     402 kB            Clients.ViewClients  [emitted]  [big]  Clients.ViewClients
                Clients.ViewScope.fe39534f812fbf965e08.v1.js     402 kB              Clients.ViewScope  [emitted]  [big]  Clients.ViewScope
               Clients.ViewScopes.fc0fcaa791482f6c5d6c.v1.js     390 kB             Clients.ViewScopes  [emitted]  [big]  Clients.ViewScopes
                        Dashboard.bc8aee4b1087982c6822.v1.js     157 kB                      Dashboard  [emitted]         Dashboard
                    Documentation.83e7e4424fb3095dd05f.v1.js     150 kB                  Documentation  [emitted]         Documentation
                             Home.761ea8d2dfaf1f55a790.v1.js    46.5 kB                           Home  [emitted]         Home
                            Hooks.3bc2497a3981ecbf9cd4.v1.js    1.43 kB                          Hooks  [emitted]         Hooks
                  Hooks.ListHooks.03c242b05de3096b69d3.v1.js     312 kB                Hooks.ListHooks  [emitted]  [big]  Hooks.ListHooks
                   Hooks.ViewHook.29b659223ee4d803aec9.v1.js     788 kB                 Hooks.ViewHook  [emitted]  [big]  Hooks.ViewHook
                         NotFound.eb0ae5693d261f1a08b5.v1.js     152 kB                       NotFound  [emitted]         NotFound
                     Provisioners.f82dc17adc223d3b398b.v1.js    1.89 kB                   Provisioners  [emitted]         Provisioners
    Provisioners.ViewProvisioners.cb190bff1ea55182656b.v1.js     269 kB  Provisioners.ViewProvisioners  [emitted]  [big]  Provisioners.ViewProvisioners
          Provisioners.ViewWorker.3aabda3bac385492d9f0.v1.js     478 kB        Provisioners.ViewWorker  [emitted]  [big]  Provisioners.ViewWorker
     Provisioners.ViewWorkerTypes.167f2f8476eb3571911d.v1.js     417 kB   Provisioners.ViewWorkerTypes  [emitted]  [big]  Provisioners.ViewWorkerTypes
         Provisioners.ViewWorkers.a2da3b54fcb60754f54f.v1.js     442 kB       Provisioners.ViewWorkers  [emitted]  [big]  Provisioners.ViewWorkers
                            Roles.985f69e935c75ed84d5a.v1.js    1.42 kB
```

The build with file specific imports seems to yield a smaller build. We can go with that one.